core/bessd.cc

```diff
@@ -360,5 +362,47 @@ bool SetResourceLimit() {
   }
 }
 
+static inline bool has_suffix(const std::string &s, const std::string &suffix) {
+  return (s.size() >= suffix.size()) &&
+         std::equal(suffix.rbegin(), suffix.rend(), s.rbegin());
+}
+
+bool LoadModule(const std::string &module_path) {
+  return (dlopen(module_path.c_str(), RTLD_NOW) != nullptr);
+}
+
+bool LoadModules(const std::string &directory) {
+  DIR *dir = opendir(directory.c_str());
+  if (!dir) {
+    return false;
+  }
+  dirent *entry;
+  while ((entry = readdir(dir)) != nullptr) {
+    if (has_suffix(entry->d_name, ".so")) {
+      const std::string full_path = directory + "/" + entry->d_name;
+      LOG(INFO) << "Loading module: " << full_path;
+      if (!LoadModule(full_path)) {
+        LOG(WARNING) << "Cannot load module " << full_path;
+        closedir(dir);
+        return false;
+      }
+    }
+  }
+  closedir(dir);
+  return true;
+}
+
+std::string GetCurrentDirectory() {
+  char dest[PATH_MAX];
+  if (readlink("/proc/self/exe", dest, PATH_MAX) == -1) {
```

Comment:
readlink() doesn't guarantee null termination, so dest could either fill up to PATH_MAX or not but still not have a '\0' at the end, and strrchr() won't be able to find the end.